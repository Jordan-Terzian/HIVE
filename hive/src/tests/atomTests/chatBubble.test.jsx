import React from 'react';
import { render, screen } from '@testing-library/react';
import ChatBubble from '../../components/atoms/chatBubble';

describe('ChatBubble Component Tests', () => {

    it('should render the component correctly', () => {
        render(<ChatBubble sender="John" text="Hello" isSent={true} timestamp="10:00 AM" pic="url_to_pic" />);
        expect(screen.getByText('Hello')).toBeInTheDocument();
    });

    it('should display the correct text and timestamp', () => {
        render(<ChatBubble sender="John" text="Hello" isSent={false} timestamp="10:00 AM" pic="url_to_pic" />);
        expect(screen.getByText('Hello')).toBeInTheDocument();
        expect(screen.getByText('10:00 AM')).toBeInTheDocument();
    });

    it('should display the profile picture on the left for received messages', () => {
        render(<ChatBubble sender="John" text="Hello" isSent={false} timestamp="10:00 AM" pic="url_to_pic" />);
    });

    it('should display the profile picture on the right for sent messages', () => {
        render(<ChatBubble sender="John" text="Hello" isSent={true} timestamp="10:00 AM" pic="url_to_pic" />);
    });

    it('should match the snapshot', () => {
        const { asFragment } = render(<ChatBubble sender="John" text="Hello" isSent={true} timestamp="10:00 AM" pic="url_to_pic" />);
        expect(asFragment()).toMatchSnapshot();
    });

});