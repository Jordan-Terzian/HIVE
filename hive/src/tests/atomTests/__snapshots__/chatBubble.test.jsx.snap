// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBubble Component Tests should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-11i7foc"
  >
    <div
      class="MuiBox-root css-1x1x1eb"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
      >
        Hello
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption css-xyekd0-MuiTypography-root"
      >
        10:00 AM
      </span>
    </div>
    <div
      class="MuiBox-root css-1mugvbl"
    >
      <img
        alt="John's profile"
        src="url_to_pic"
        style="width: 100%; height: 100%;"
      />
    </div>
  </div>
</DocumentFragment>
`;
